packages/forms-collection/src/new-app/_form.tsx

```diff
@@ -68,6 +72,65 @@ export const AppForm = ({
     }
   };
 
+  // Auto-select icon based on app name with debounced search
+  const { data: iconsData } = clientApi.icon.findIcons.useQuery(
+    {
+      searchText: debouncedName,
+    },
+    {
+      enabled: debouncedName.length > 3,
+    },
+  );
+
+  useEffect(() => {
+    if (debouncedName && !form.values.iconUrl && iconsData?.icons) {
+      // Find exact matches first, then partial matches
+      let exactSvgMatch = null;
+      let exactNonSvgMatch = null;
+      let partialSvgMatch = null;
+      let partialNonSvgMatch = null;
+
+      const nameLower = debouncedName.toLowerCase();
+
+      for (const group of iconsData.icons) {
+        for (const icon of group.icons) {
+          const iconNameLower = icon.url.toLowerCase();
+          const fileNameParts = iconNameLower.split("/");
+          const fileName = fileNameParts[fileNameParts.length - 1]?.split(".")[0];
+
+          // Check for exact match first (the file name equals the search term)
+          if (fileName === nameLower) {
+            if (icon.url.endsWith(".svg")) {
+              exactSvgMatch = icon.url;
+            } else if (!exactNonSvgMatch) {
+              exactNonSvgMatch = icon.url;
+            }
+          }
+          // Then check for partial match
+          else if (fileName?.includes(nameLower)) {
+            if (icon.url.endsWith(".svg") && !partialSvgMatch) {
+              partialSvgMatch = icon.url;
+            } else if (!partialNonSvgMatch) {
+              partialNonSvgMatch = icon.url;
+            }
+          }
+        }
+        if (exactSvgMatch) break;
+      }
+
+      // Set the icon URL with priority: exact SVG > exact non-SVG > partial SVG > partial non-SVG
+      if (exactSvgMatch) {
+        form.setFieldValue("iconUrl", exactSvgMatch);
+      } else if (exactNonSvgMatch) {
+        form.setFieldValue("iconUrl", exactNonSvgMatch);
+      } else if (partialSvgMatch) {
+        form.setFieldValue("iconUrl", partialSvgMatch);
+      } else if (partialNonSvgMatch) {
+        form.setFieldValue("iconUrl", partialNonSvgMatch);
+      }
+    }
```

Comment:
I have a few issues with this code:
- The code is repetitive (e.g. ``setFieldValue``)
- The prioritization could be implemented with less code duplication
- We iterate over **all** icons, which will lead to poor performance. We should interrupt the loop when we find a match

Reply:
You're right, I added an early exit when there is an exact match and removed all of the variables